### PR TITLE
Add Sara Kothari submission

### PR DESCRIPTION
**Pull request by sara-kothari**
Final validation loss: 3.36327

Learning curve: https://api.wandb.ai/links/sarako-stanford-university/iop3e8la

What I did:
1. Compared to the baseline configuration, I increased model depth (4 to 8 layers),
halved the batch size (256 to 128), and tuned the learning rate schedule (peak LR,
warmup, and decay parameters).
2. I also used "torch.set_float32_matmul_precision('high')". This led to a 3.5x speedup.
3. I used torch.compile
4. I tied the weights of the input and output embeddings together
5. I changed d_model to 768 and d_ff to 2048

### CODE REVIEW

**Comment by marcelroed** [comment]
Looks good, thank you!
